### PR TITLE
Remove argument from parseExpression: and parseMethod: in Parser

### PR DESCRIPTION
**Pull request by dionisiydk**
Little simplification of Parser:
Remove argument from methods #parseExpression: and #parseMethod:  in Parser (instance side). Use instance variable source instead of argument (parser is always prepared with the source code in existing use cases).

### CODE REVIEW

**Comment by dionisiydk** [comment]
Failed tests are unrelated

**Comment by estebanlm** [comment]
hi @dionisiydk, can you add a `parseExpression: aString` and `parseMethod: aString` deprecated methods?
this is the kind of change that breaks a lot of tools ;)

**Comment by dionisiydk** [comment]
I will add deprecations. But those methods were really private considering
the implementation.
What exact tools become broken? So I can check .

вт, 18 мая 2021 г. в 08:27, Esteban Lorenzano ***@***.***>:

**Comment by estebanlm** [comment]
is a generic comment, people doing tools can use it and we cannot assume is private unless is in `private` protocol (which is not the case, maybe it worth move them there?) :)

in freeze stage, I do not want to introduce unnecessary noise ;)

File: src/AST-Core/RBParser.class.st [hunk 1]
```diff
@@ -750,21 +749,10 @@ RBParser >> parseMethod [
...
-	self atEnd
-		ifFalse: [ 
-			 ^ self parseIncompleteExpression: node body ].
-	node source: aMethodDefinitionString.
-	^ node
+	methodNode source: source.
+	self atEnd ifFalse: [ ^ self parseIncompleteExpression: methodNode body ]. 
+	methodNode source: source.
+	^ methodNode
 ]
```
Comment (dionisiydk) [inline src/AST-Core/RBParser.class.st:756]:
```suggestion
]

{ #category : #accessing }
RBParser >> parseMethod: aMethodDefinitionString [
	self 
		deprecated: 'Use #parseMethod without argument instead.' 
		transformWith: '`@receiver parseMethod: `@arg1 ' -> '`@receiver parseMethod'. 
	
	^self parseMethod
]
```

File: src/AST-Core/RBParser.class.st [hunk 2]
```diff
@@ -1220,7 +1208,10 @@ RBParser >> parseUnaryPattern [
 	| selector |
 	selector := currentToken.
 	self step.
-	^self methodNodeClass selector: selector value asSymbol keywordsPositions: (Array with: selector start) arguments: #()
+	^self methodNodeClass 
+		selector: selector value asSymbol 
+		keywordsPositions: (Array with: selector start) 
+		arguments: #()
```
Comment (dionisiydk) [inline src/AST-Core/RBParser.class.st:1214]:
```suggestion
		arguments: #()
]

{ #category : #accessing }
RBParser >> parseMethod: aMethodDefinitionString [
	self 
		deprecated: 'Use #parseMethod without argument instead.' 
		transformWith: '`@receiver parseMethod: `@arg1 ' -> '`@receiver parseMethod'. 
	
	^self parseMethod
```		

```

File: src/AST-Core/RBParser.class.st [hunk 3]
```diff
@@ -750,21 +749,10 @@ RBParser >> parseMethod [
...
-	node source: aMethodDefinitionString.
-	self atEnd
-		ifFalse: [ 
-			 ^ self parseIncompleteExpression: node body ].
-	node source: aMethodDefinitionString.
-	^ node
+	methodNode source: source.
+	self atEnd ifFalse: [ ^ self parseIncompleteExpression: methodNode body ]. 
+	methodNode source: source.
+	^ methodNode
```
Comment (dionisiydk) [inline src/AST-Core/RBParser.class.st:755]:
```suggestion
	^ methodNode
]

{ #category : #accessing }
RBParser >> parseMethod: aMethodDefinitionString [
	self 
		deprecated: 'Use #parseMethod without argument instead.' 
		transformWith: '`@receiver parseMethod: `@arg1 ' -> '`@receiver parseMethod'. 
	
	^self parseMethod
]
```
```

**Comment by dionisiydk** [comment]
> is a generic comment, people doing tools can use it and we cannot assume is private unless is in `private` protocol (which is not the case, maybe it worth move them there?) :)

Actually #parseMethod: was in private-parsing protocol but #parseExpression: was in accessing.
Anyway I will add deprecations

**Comment by estebanlm** [comment]
lol, maybe both should be there... but anyway. Deprecations now, removal in P10 ;)

**Review by estebanlm — APPROVED** [review]

**Pull request merged**